src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -223,22 +229,20 @@ missingAlternative env mn ca uncovered
   mcases = missingCases env mn uncovered ca
 
 -- |
--- Main exhaustivity checking function
+-- Main exhaustivity checking function.
 -- Starting with the set `uncovered = { _ }` (nothing covered, one `_` for each function argument),
 -- it partitions that set with the new uncovered cases, until it consumes the whole set of clauses.
 -- Then, returns the uncovered set of case alternatives.
 -- 
 checkExhaustive :: forall m. (MonadWriter MultipleErrors m) => Environment -> ModuleName -> [CaseAlternative] -> m ()
-checkExhaustive env mn cas = makeResult . first nub $ foldl' step ([initial], (pure True, [])) cas
+checkExhaustive env mn cas = makeResult . first (nub . take maxCases) $ foldr step ([initial], (pure True, [])) cas
```

Comment:
Since we've hit the pencils down date for GSOC, don't worry about spending too much time on this. I'd rather spend the remaining time implementing tests for your `ecl` library and getting it published, as well as a wrap-up blog post.